compiler/src/iree/compiler/Codegen/Common/OptimizeTensorInsertExtractSlices.cpp

```diff
@@ -84,63 +84,73 @@ hoistLoopInvariantSubsetAtIterArg(RewriterBase &rewriter,
     return loopLike;
   }
 
-  // Get all subset extraction uses of this iter_arg and try to hoist them
-  // out of the loop.
-  for (Operation *op : loopLike.getRegionIterArgs()[idx].getUsers()) {
-    auto extraction = dyn_cast<SubsetExtractionOpInterface>(op);
-    if (!extraction) {
-      continue;
-    }
+  bool changed = true;
+  while (changed) {
```

Comment:
Will review more when not on a phone, but this feels like a usecase for `do { ... } while(changed)`